bundles/org.openhab.binding.solax/src/main/java/org/openhab/binding/solax/internal/SolaxLocalAccessHandler.java

```diff
@@ -90,68 +101,192 @@ private void retrieveData() {
-                new QuantityType<>(data.getPVTotalCurrent(), Units.AMPERE));
+        updateState(SolaxBindingConstants.INVERTER_OUTPUT_VOLTAGE_PHASE1,
+                new QuantityType<>(threePhaseData.getVoltagePhase1(), Units.VOLT));
+        updateState(SolaxBindingConstants.INVERTER_OUTPUT_VOLTAGE_PHASE2,
+                new QuantityType<>(threePhaseData.getVoltagePhase2(), Units.VOLT));
+        updateState(SolaxBindingConstants.INVERTER_OUTPUT_VOLTAGE_PHASE3,
+                new QuantityType<>(threePhaseData.getVoltagePhase3(), Units.VOLT));
 
-        updateState(SolaxBindingConstants.BATTERY_POWER, new QuantityType<>(data.getBatteryPower(), Units.WATT));
-        updateState(SolaxBindingConstants.BATTERY_CURRENT, new QuantityType<>(data.getBatteryCurrent(), Units.AMPERE));
-        updateState(SolaxBindingConstants.BATTERY_VOLTAGE, new QuantityType<>(data.getBatteryVoltage(), Units.VOLT));
-        updateState(SolaxBindingConstants.BATTERY_TEMPERATURE,
-                new QuantityType<>(data.getBatteryTemperature(), SIUnits.CELSIUS));
-        updateState(SolaxBindingConstants.BATTERY_STATE_OF_CHARGE,
-                new QuantityType<>(data.getBatterySoC(), Units.PERCENT));
+        updateState(SolaxBindingConstants.INVERTER_OUTPUT_FREQUENCY_PHASE1,
+                new QuantityType<>(threePhaseData.getFrequencyPhase1(), Units.HERTZ));
+        updateState(SolaxBindingConstants.INVERTER_OUTPUT_FREQUENCY_PHASE2,
+                new QuantityType<>(threePhaseData.getFrequencyPhase2(), Units.HERTZ));
+        updateState(SolaxBindingConstants.INVERTER_OUTPUT_FREQUENCY_PHASE3,
+                new QuantityType<>(threePhaseData.getFrequencyPhase3(), Units.HERTZ));
+    }
 
-        updateState(SolaxBindingConstants.FEED_IN_POWER, new QuantityType<>(data.getFeedInPower(), Units.WATT));
+    private void removeUnsupportedChannels(Set<String> supportedChannels) {
+        if (supportedChannels.isEmpty()) {
+            return;
+        }
+        List<Channel> channels = getThing().getChannels();
+        List<Channel> channelsToRemove = channels.stream()
+                .filter(channel -> !supportedChannels.contains(channel.getUID().getId())).collect(Collectors.toList());
 
-        updateState(SolaxBindingConstants.TIMESTAMP, new DateTimeType(ZonedDateTime.now()));
-        updateState(SolaxBindingConstants.RAW_DATA, new StringType(data.getRawData()));
+        if (!channelsToRemove.isEmpty()) {
+            if (logger.isDebugEnabled()) {
+                logRemovedChannels(channelsToRemove);
+            }
+            updateThing(editThing().withoutChannels(channelsToRemove).build());
+        }
+    }
+
+    private void logRemovedChannels(List<Channel> channelsToRemove) {
+        List<String> channelsToRemoveForLog = channelsToRemove.stream().map(channel -> channel.getUID().getId())
+                .collect(Collectors.toList());
+        logger.debug("Detected not supported channels for the current i. Channels to be removed:{}",
```

Comment:
What does "current i" mean?

Reply:
"current i" means "current inverter" obviously :)))) (will be fixed in next commit)